src/code/audio/code_801A5BD0.c

```diff
@@ -1,45 +1,835 @@
+        Audio_SetVolumeScale(SEQ_PLAYER_BGM_MAIN, 2, 0x7F, 0xF);
+        Audio_SetVolumeScale(SEQ_PLAYER_BGM_SUB, 2, 0x7F, 0xF);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/code_801A5BD0/func_801A7828.s")
+void Audio_PlaySfxGeneral(u16 sfxId, Vec3f* pos, u8 token, f32* freqScale, f32* vol, s8* reverbAdd) {
+    u8 i;
+    SfxRequest* reqWrite;
+    SfxRequest* reqRead;
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/code_801A5BD0/func_801A787C.s")
+    if (!gSfxBankMuted[SFX_BANK_SHIFT(sfxId)]) {
+        reqWrite = &sSfxRequests[sSfxRequestWriteIndex];
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/code_801A5BD0/Audio_IsSfxPlaying.s")
+        for (i = sSfxRequestReadIndex; sSfxRequestWriteIndex != i; i++) {
+            reqRead = &sSfxRequests[i];
+            if (reqRead->pos == pos && reqRead->sfxId == sfxId) {
+                return;
+            }
+        }
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/code_801A5BD0/func_801A794C.s")
+        reqWrite->sfxId = sfxId;
+        reqWrite->pos = pos;
+        reqWrite->token = token;
+        reqWrite->freqScale = freqScale;
+        reqWrite->vol = vol;
+        reqWrite->reverbAdd = reverbAdd;
+        sSfxRequestWriteIndex++;
+    }
+}
+
+void Audio_RemoveMatchingSfxRequests(u8 aspect, SfxBankEntry* entry) {
+    SfxRequest* req;
+    s32 remove;
+    u8 i = sSfxRequestReadIndex;
+
+    for (; i != sSfxRequestWriteIndex; i++) {
+        remove = false;
+        req = &sSfxRequests[i];
+
+        switch (aspect) {
+            case 0:
+                if (SFX_BANK_MASK(req->sfxId) == SFX_BANK_MASK(entry->sfxId)) {
+                    remove = true;
+                }
+                break;
+            case 1:
+                if ((SFX_BANK_MASK(req->sfxId) == SFX_BANK_MASK(entry->sfxId)) && (&req->pos->x == entry->posX)) {
+                    remove = true;
+                }
+                break;
+            case 2:
+                if (&req->pos->x == entry->posX) {
+                    remove = true;
+                }
+                break;
+            case 3:
+                if ((&req->pos->x == entry->posX) && (req->sfxId == entry->sfxId)) {
+                    remove = true;
+                }
+                break;
+            case 4:
+                if ((req->token == entry->token) && (req->sfxId == entry->sfxId)) {
+                    remove = true;
+                }
+                break;
+            case 5:
+                if (req->sfxId == entry->sfxId) {
+                    remove = true;
+                }
+                break;
+        }
+
+        if (remove) {
+            req->sfxId = 0;
+        }
+    }
+}
+
+void Audio_ProcessSfxRequest(void) {
+    u16 sfxId;
+    u8 count;
+    u8 index;
+    SfxRequest* req;
+    SfxBankEntry* entry;
+    SfxParams* sfxParams;
+    s32 bankId;
+    u8 evictImportance;
+    u8 evictIndex;
+
+    req = &sSfxRequests[sSfxRequestReadIndex];
+    evictIndex = 0x80;
+
+    if (req->sfxId == 0) {
```

Comment:
Would it be better this way?
```suggestion
    u8 index;
    SfxRequest* req = &sSfxRequests[sSfxRequestReadIndex];
    SfxBankEntry* entry;
    SfxParams* sfxParams;
    s32 bankId;
    u8 evictImportance;
    u8 evictIndex = 0x80;

    if (req->sfxId == 0) {
```